solutions/USACO open18-multiplayermoo.mdx

```diff
@@ -0,0 +1,130 @@
+---
+id: usaco-836
+source: USACO Silver 2018 US Open
+title: Multiplayer Moo
+author: Jesse Choe
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_multimoo_silver_open18.html)
+
+[Video Analysis (Java) ](https://www.youtube.com/watch?v=dCY5EOKyboM)
+
+This problem is an example of a "floodfill" problem. This problem is best solved when divided into two subproblems.
+
+The first subproblem is to find the maximum number of adjacent cells comprising of the same ID number. To calculate the maximum number of adjacent cells, we can iterate over each cell in the grid and count the maximum number of "connected" cells via a recursive floodfill approach. To avoid recomputation, we can store a boolean array of the cells we "visited" and only run the floodfill on this cell if and only if that cell hasn't been visited.
+
+The second subproblem is to find the maximum number of adjacent cells comprising of exactly two distinct ID numbers (note: if two different ID numbers are adjacent to each other, then they are considered "connected"). To calculate the maximum number of "connected" cells, we can iterate over each cell in the grid and compute this by choosing any two adjacent cells as the ID's. From here, we can run a floodfill and find the maximum number of "connected" cells over all possible cells. As used in the previous subproblem, we can use a boolean array to prevent recomputation.
+
+**Time Complexity:** $\mathcal{O}(N^4)$
```

Comment:
this solution isn't supposed to pass though ... (it's pretty easy to break)

Reply:
Well idk, it passes all the USACO tcs and the official solution is way too ugly.

Reply:
> this solution isn't supposed to pass though ... (it's pretty easy to break)

Does this mean that every singe test case on USACO was bad?

Reply:
yes ...

Reply:
So we can't put this sol despite passing 10/10 tcs?

Reply:
yes

Reply:
:(